### PR TITLE
Add support to build psa-arch-test protected storage test binary

### PR DESCRIPTION
**Pull request by simonqhughes**
This PR represents the submission of work-in-progress for building the psa-arch-tests protected storage test binary. It includes:
- A repo manifest for creating a workspace containing projects needed to build the test binary.
- A makefile for building the test binary.
- mbed-crypto patches for patching a specific version so the mbed-crypto tests can be leveraged for testing psa_trusted_storage.

This work was carried out under IOTMBL-2532.

### CODE REVIEW

**Comment by simonqhughes** [comment]
@jjohnson-arm Please could you take a look at this PR for iotmbl-2532? There is a follow-on PR for a pinned manifest that can only be created after this PR has been merged.

**Comment by simonqhughes** [comment]
@jainvikas8 Hi, would you be able to review this PR for me? There are instructions in the test/Makefile 
 for how to build ps-arch-test-ps here: https://github.com/ARMmbed/psa_trusted_storage_linux/pull/9/files#diff-0d8f5a336dfc7f93d0f8ebb4026e4d46

You need to install the repo tool. Let me know if you need help with that.

**Comment by jainvikas8** [comment]
@simonqhughes  I have run the test with `make test-psa-arch-tests-ps` after downloading the repo `make -f proj_mak ws-create`

There are 2 tests failing (406 and 408)
* If we could investigate regarding the failure and raise issue on [psa-arch-tests](https://github.com/ARM-software/psa-arch-tests) or fix as part of the PR, that would be good.

**Comment by simonqhughes** [comment]
As requested, issue for Test 406 created here:

https://github.com/ARM-software/psa-arch-tests/issues/125

**Comment by simonqhughes** [comment]
@jainvikas8 I've put additional changes on this PR to fix the psa-arch-tests-ps test failures. Test 406 and 408 are now working. Please could you review and hopefully approve? Thanks.

File: lib/common_storage.h [hunk 1]
```diff
@@ -10,6 +10,10 @@
 #include "psa/error.h"
 #include "psa/storage_common.h"
 
+/* IHI0087-PS_-Storage_API_1.0.0.pdf specifies 0 to be an invalid uid value
+ * and should be treated as a nerror */
```
Comment (jainvikas8) [inline lib/common_storage.h]: ***as an error***

**Review by jainvikas8 — APPROVED** [review]

**Pull request merged**